### PR TITLE
Update Contributing to containing the one to one relation information

### LINKED ISSUE #389: Update Contriibuting.md to explain 1:1 ratio between extensions and tests
Every SwifterSwift extension (properties, methods, inits) should have its own separate test.

### PR DESCRIPTION
**Pull request by BennX**
Update Contributing to containing the one to one relation between extension and test file and added some note that every extension should be tested.

Should resolve #389

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -67,8 +67,12 @@ public extension SomeType {
 	- Properties & static properties
 	- Methods & static methods
 	- Initializers
-- Files are named based on the type that the contained extensions extend (example: all String extensions are found in "**StringExtensions.swift**" file)
-- extensions are ordered inside files in the following order:
+- Files are named based on the type that the contained extensions extend. (example: all String extensions are found in "**StringExtensions.swift**" file)
+- All extensions should be tested.
+- All extensions files and test file have a one to one relation.
```
Comment (calebkleveter) [inline CONTRIBUTING.md]: Looks like you might have a typo here. Should it be "test files"? You might also want to explain what it means for the files to have a one to one relation, but it could go either way ¯\_(ツ)_/¯.

Comment (BennX) [reply CONTRIBUTING.md]: Agree on the type and explanation. I wasn't sure since it's more or less explained right in the next line but ¯\_(ツ)_/¯

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -67,8 +67,12 @@ public extension SomeType {
 	- Properties & static properties
 	- Methods & static methods
 	- Initializers
-- Files are named based on the type that the contained extensions extend (example: all String extensions are found in "**StringExtensions.swift**" file)
-- extensions are ordered inside files in the following order:
+- Files are named based on the type that the contained extensions extend. (example: all String extensions are found in "**StringExtensions.swift**" file)
+- All extensions should be tested.
+- All extensions files and test files have a one to one relation.(example: all tests for "**StringExtensions.swift**" are found in the "**StringExtensionsTests.swift**" file)
+- All test files are named based on the extensions which it tests. (example: all String extensions tests are found in "**StringExtensionsTests.swift**" file)
+- An single extension can have multiple tests.
```
Comment (LucianoPAlmeida) [inline CONTRIBUTING.md]: I think here we have to explain the one to one extension-test methods relation. Example: the extension Array.rotate() should have one testRotate() test on ArrayTests file where all the edge cases are for this are tested inside this test :)

**Comment by BennX** [comment]
@LucianoPAlmeida had some spare time. I actually disagree on having only one test for all extensions. More complex might require more then one test but i agree on that those kind of extem complex extension may not be wanted within the Framework or can have more then one test.

**Comment by LucianoPAlmeida** [comment]
Hey @BennX I think I explained wrongly, it's actually one test for extension method not for all extensions in the file :) sorry for that 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by LucianoPAlmeida** [comment]
@BennX Sorry  I think I missread 😂
I actually agree with you :) For more complex extensions this rule may not apply.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -67,8 +67,12 @@ public extension SomeType {
 	- Properties & static properties
 	- Methods & static methods
 	- Initializers
-- Files are named based on the type that the contained extensions extend (example: all String extensions are found in "**StringExtensions.swift**" file)
-- extensions are ordered inside files in the following order:
+- Files are named based on the type that the contained extensions extend. (example: all String extensions are found in "**StringExtensions.swift**" file)
+- All extensions should be tested.
+- All extensions files and test files have a one to one relation.(example: all tests for "**StringExtensions.swift**" are found in the "**StringExtensionsTests.swift**" file)
+- All extensions have a one to one relation to a test which should cover all edge cases and is named like the extension. (example: `DateExtensions` method `isBetween` is tested in file: `DateExtensionsTest` test: `testIsBetween`)
```
Comment (SD10) [inline CONTRIBUTING.md]: Can we split this into two points?
- There should be a one to one relationship between extensions and their backing tests.
- Tests should be named using the same API of the extension it backs.
(example: `DateExtensions` method `isBetween` is named `testIsBetween`)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -67,8 +67,12 @@ public extension SomeType {
...
+- All test files are named based on the extensions which it tests. (example: all String extensions tests are found in "**StringExtensionsTests.swift**" file)
+- extensions and tests are ordered inside files in the following order:
```
Comment (SD10) [inline CONTRIBUTING.md]: Capitalize `extensions` please 😝

**Comment by SD10** [comment]
This is super useful @BennX. I just have some minor changes 👍

**Comment by SD10** [comment]
I think we should also ask users to submit only 1 new extension per PR 😅

**Comment by BennX** [comment]
@SD10 should be done 👍 I added the PR note to to the end. Looks like the best spot to me.

**Comment by SD10** [comment]
Thanks again @BennX 👍
